core/src/main/java/com/linecorp/armeria/internal/client/DecodedHttpResponse.java

```diff
@@ -79,8 +79,12 @@ protected void onRemoval(HttpObject obj) {
         if (obj instanceof HttpData) {
             final int length = ((HttpData) obj).length();
             assert inboundTrafficController != null;
-            assert id > -1 : "id must be set before consuming HttpData";
-            inboundTrafficController.dec(id, length);
+            assert streamId > -1 : "id must be set before consuming HttpData";
+            if (streamId == 1) {
+                // The stream ID 1 is used for an HTTP/1 upgrade request.
```

Comment:
If we want to handle this as a special case, I think we should also do it in `tryWrite`.

Reply:
Makes sense. Done.